build.gradle.kts

```diff
@@ -1,137 +1,271 @@
-        freeCompilerArgs.addAll(
-            "-version",
-            "-Xjsr305=strict",
-            "-opt-in=kotlin.RequiresOptIn",
-        )
-        verbose.set(true)
+idea {
+    module {
+        // Marks the functionTest as a test source set
+        testSources.from(sourceSets.functionalTest.get().allSource.srcDirs)
     }
 }
 
+kotlin {
+    jvmToolchain(17)
+}
+
 java {
     withSourcesJar()
     withJavadocJar()
 }
 
-tasks.withType<Test>().configureEach {
-    useJUnitPlatform()
-    testLogging {
-        showStandardStreams = true
-        showCauses = true
-        showStackTraces = true
-        events = TestLogEvent.values().toSet()
-        exceptionFormat = TestExceptionFormat.FULL
+detekt {
+    buildUponDefaultConfig = true
+    allRules = false
+    config.from(files("detekt.yml"))
+}
+
+spotless {
+    format("misc") {
+        target("*.md", "*.gitignore", "*.yml", "*.yaml", "*.toml", "*.properties")
+        trimTrailingWhitespace()
+        endWithNewline()
+    }
+
+    kotlin {
+        target("src/**/*.kt")
+        ktlint()
+        trimTrailingWhitespace()
+        endWithNewline()
+        licenseHeaderFile(rootProject.file("spotless/license.kt"))
+    }
+
+    kotlinGradle {
+        target("*.kts", "src/**/*.kts")
+        ktlint()
+        trimTrailingWhitespace()
+        endWithNewline()
+        licenseHeaderFile(
+            rootProject.file("spotless/license.kt"),
+            "(import|plugins|buildscript|dependencies|pluginManagement|dependencyResolutionManagement)",
+        )
     }
 }
 
-// project version, also used for publishing
-version = semver.version
+testlogger {
+    theme = ThemeType.STANDARD
+    showCauses = true
+    slowThreshold = 1000
+    showSummary = true
+    showStandardStreams = false
+}
 
-val githubTokenValue: String = findProperty("githubToken")?.toString() ?: System.getenv("GITHUB_TOKEN")
+apiValidation {
+    ignoredPackages += listOf(
+        // Internal package is not part of the public API
+        "$group.internal",
+    )
+}
 
-githubRelease {
-    apiEndpoint = "https://api.github.com" // should only change for GitHub enterprise users
-    body = ""
-    draft = false
-    dryRun = false // by default false; you can use this to see what actions would be taken without making a release
-    generateReleaseNotes = true
-    overwrite = false
-    owner = "FigureTechnologies"
-    prerelease = false
-    repo = "gradle-semver-plugin" // Should automatically pick up your project name, overwrite this if you need something else
-    targetCommitish = System.getenv("TARGET_COMMIT_ISH")?.takeIf { it.isNotBlank() } ?: "main"
+class PublishingConstants {
+    val name = "Gradle Semver Plugin"
+    val description = "Gradle Plugin for Automatic Semantic Versioning"
+    val pluginImplementation = "$group.SemverPlugin"
+    val tags = listOf("semver", "versioning", "git")
+
+    val website = "https://figuretechnologies.github.io/gradle-semver-plugin"
```

Comment:
I currently get a 404 for this URL

Reply:
@jonasg13 you will until this gets merged in and the documentation gets generated on the `gh-pages` branch.